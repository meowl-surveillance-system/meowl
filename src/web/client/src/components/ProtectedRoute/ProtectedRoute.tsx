import * as React from "react";
import { Redirect, Route, RouteProps } from "react-router";

export interface Props extends RouteProps {
  isLoggedIn: boolean;
  redirectPath: string;
}

export default class ProtectedRoute extends Route<Props> {
  render() {
    console.log(this.props.isLoggedIn);

    if (this.props.isLoggedIn === false) {
      return <Redirect to={{ pathname: this.props.redirectPath }} />;
    } else {
      return <Route {...this.props} />;
    }
  }
}
