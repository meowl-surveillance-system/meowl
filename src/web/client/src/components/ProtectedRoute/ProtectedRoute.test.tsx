import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import ProtectedRoute from "./ProtectedRoute";

describe("ProtectedRoute component", () => {
  test("renders ProtectedRoute component", () => {
    const { container } = render(
      <Router>
        <ProtectedRoute isLoggedIn={false} redirectPath={"/"} />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
