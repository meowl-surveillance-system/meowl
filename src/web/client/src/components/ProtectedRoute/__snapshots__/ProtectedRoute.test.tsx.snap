// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute component renders ProtectedRoute component 1`] = `null`;
